pinecone/core/grpc/index_grpc.py

```diff
@@ -264,37 +297,141 @@ def _vector_transform(item):
+            async_req (bool): If True, the delete operation will be performed asynchronously.
+                              Defaults to False. [optional]
+
+        Returns: DeleteResponse (contains no data) or a PineconeGrpcFuture object if async_req is True.
+
+        Call Examples:
+            index.delete(ids=['id1', 'id2'], namespace='my_namespace')
+            index.delete(delete_all=True, namespace='my_namespace')
+            index.delete(filter={'key': 'value'}, namespace='my_namespace', async_req=True)
+        """
+
+        if filter is not None:
+            filter = dict_to_proto_struct(filter)
+
+        args_dict = self._parse_args_to_dict([('ids', ids),
+                                              ('delete_all', delete_all),
+                                              ('namespace', namespace),
+                                              ('filter', filter)])
+
+        request = DeleteRequest(**args_dict, **kwargs)
         timeout = kwargs.pop('timeout', None)
         if async_req:
             future = self._wrap_grpc_call(self.stub.Delete.future, request, timeout=timeout)
             return PineconeGrpcFuture(future)
         else:
             return self._wrap_grpc_call(self.stub.Delete, request, timeout=timeout)
 
-    def fetch(self, *args, **kwargs):
+    def fetch(self,
+              ids: Optional[List[str]],
+              namespace: Optional[str] = None,
+              **kwargs) -> FetchResponse:
+        """
+        The fetch operation looks up and returns vectors, by ID, from a single namespace.
+        The returned vectors include the vector data and/or metadata.
+
+        Args:
+            ids (List[str]): The vector IDs to fetch.
+            namespace (str): The namespace to fetch vectors from.
+                             If not specified, the default namespace is used. [optional]
+
+        Returns: FetchResponse object which contains the list of Vector objects, and namespace name.
+
+        Call Examples:
+            index.fetch(ids=['id1', 'id2'], namespace='my_namespace')
+            index.fetch(ids=['id1', 'id2'])
+        """
         timeout = kwargs.pop('timeout', None)
-        request = FetchRequest(*args, **kwargs)
+
+        args_dict = self._parse_args_to_dict([('namespace', namespace)])
+
+        request = FetchRequest(ids=ids, **args_dict, **kwargs)
         response = self._wrap_grpc_call(self.stub.Fetch, request, timeout=timeout)
         json_response = json_format.MessageToDict(response)
         return parse_fetch_response(json_response)
 
-    def query(self, vector=[], id='', queries=[], **kwargs):
-        timeout = kwargs.pop('timeout', None)
-
+    def query(self,
+              vector: Optional[List[float]] = None,
+              id: Optional[str] = None,
+              queries: Optional[Union[List[GRPCQueryVector], List[Tuple]]] = None,
+              namespace: Optional[str] = None,
+              top_k: Optional[int] = None,
```

Comment:
top_k is not optional

Reply:
You are right, but since until now it was in the kwargs, I didn't have a good place to put it as arg without breaking backwards compatibility (and we decided that this PR should preserve backwards compatibility). Do you have an idea how to make it a required arg?